lib/crontab/cron_expression/parser.ex

```diff
@@ -274,6 +334,28 @@ defmodule Crontab.CronExpression.Parser do
     end
   end
 
+  @spec clean_integer_within_range(binary, binary, Range.t()) ::
+          {:ok, CronExpression.value()} | {:error, binary}
+  defp clean_integer_within_range(value, field_name, valid_values) do
+    case Integer.parse(value, 10) do
+      {number, _} ->
```

Comment:
We should not allow any remainder from parsing here. Otherwise, it was not a valid input.

```suggestion
      {number, ""} ->
```

Reply:
Good catch! I'll add a test highlighting this issue.